src/components/EmailNotifications/VerifyEmailAddress.js

```diff
@@ -0,0 +1,251 @@
+import React, { useCallback, useState } from 'react'
+import styled from 'styled-components'
+import {
+  Button,
+  Checkbox,
+  Field,
+  GU,
+  IconCheck,
+  IconCross,
+  Link,
+  TextInput,
+  textStyle,
+  useTheme,
+} from '@aragon/ui'
+import { validateEmail } from '../../utils/notifications-utils'
+import emailIllustration from '../../assets/emailIllustration.svg'
+
+const VerifyEmailAddress = React.memo(function VerifyEmailAddress({
+  updateMode,
+  compactMode,
+  email,
+  onSubscribe,
+  onSubscribeSuccess,
+  onReSendVerificationEmail,
+  onDeleteEmail,
+  onError,
+}) {
+  const theme = useTheme()
+  const [emailAddress, setEmailAddress] = useState('')
+  const [emailInvalid, setEmailInvalid] = useState(null)
+  const [termsAccepted, setTermsAccepted] = useState(false)
+
+  const handleEmailAddressBlur = useCallback(e => {
+    const email = e.target.value
+    setEmailInvalid(!validateEmail(email))
+  }, [])
+
+  const handleEmailAddressChange = useCallback(e => {
+    const email = e.target.value
+    setEmailAddress(email)
+    if (validateEmail(email)) {
+      // Set only as valid while user typing. Use blur to set invalid
+      setEmailInvalid(false)
+    }
+  }, [])
+
+  const handleSubscribe = useCallback(async () => {
+    const { subscribedEmail, error } = await onSubscribe(emailAddress)
+
+    if (error) {
+      onError()
+    }
+    onSubscribeSuccess(subscribedEmail)
+  }, [emailAddress, onError, onSubscribe, onSubscribeSuccess])
+
+  const handleReSendEmail = useCallback(async () => {
+    const { error } = await onReSendVerificationEmail()
+
+    if (error) {
+      onError()
+    }
+  }, [onError, onReSendVerificationEmail])
+
+  return (
+    <div
+      css={`
+        display: flex;
+        flex-direction: column;
+        justify-content: center;
+        padding-top: ${3 * GU}px;
+      `}
+    >
+      <div
+        css={`
+          display: flex;
+          flex-direction: column;
+          text-align: center;
+        `}
+      >
+        <img src={emailIllustration} />
+        <span
+          css={`
+            ${textStyle('title2')};
+            margin-top: ${4 * GU}px;
+          `}
+        >
+          Verify your email address
+        </span>
+
+        <span
+          css={`
+            ${textStyle('body2')};
+            color: ${theme.surfaceContentSecondary};
+            margin-top: ${1.5 * GU}px;
+          `}
+        >
+          {`Almost there! We’ve sent a verification email to ${email}.
+          Kindly check your inbox and click the link to verify your account. ${
+            updateMode
+              ? 'Alternatively, you can update this email or delete it, if you wish to unsubscribe'
+              : ''
+          }`}
+        </span>
+        {(() => {
+          if (updateMode) {
+            const buttonWidth = compactMode
+              ? '100% '
+              : `calc((100% - ${2 * GU}px) /  2)`
```

Comment:
Nitpicking: should we move this to be inside of `ActionsButtons`, as a normal React component using `css`? That way we could also avoid the inline function here.